spark-bigquery-pushdown/pushdown_common_src/main/scala/com/google/cloud/spark/bigquery/pushdowns/WindowQuery.scala

```diff
@@ -0,0 +1,17 @@
+package com.google.cloud.spark.bigquery.pushdowns
+
+import org.apache.spark.sql.catalyst.expressions.NamedExpression
+
+case class WindowQuery(
+                        expressionConverter: SparkExpressionConverter,
+                        expressionFactory: SparkExpressionFactory,
+                        windowExpressions: Seq[NamedExpression],
+                        child: BigQuerySQLQuery,
+                        alias: String)
+  extends BigQuerySQLQuery(
+      expressionConverter,
+      expressionFactory,
+      alias,
+      children = Seq(child),
+      // Need to send in the query attributes along with window expressions to avoid "TreeNodeException: Binding attribute, tree"
+      projections = Some(child.outputWithQualifier ++ windowExpressions)) {}
```

Comment:
Where are the ordered projections? I was suggesting that we could have a companion object to WindowQuery that computes the projection. Something like

```
object WindowQuery {
  def getWindowProjections(windowExpressions: Seq[NamedExpression], child: BigQuerySQLQuery, fields: Option[Seq[Attribute]]): Option[Seq[NamedExpression]] = {
    val projectionVector = windowExpressions ++ child.outputWithQualifier

    val orderedProjections =
      fields.map(_.map(reference => {
        val origPos = projectionVector.map(_.exprId).indexOf(reference.exprId)
        projectionVector(origPos)
      }))

    orderedProjections
  }
```